Automattic/wp-calypso
# Framework: Replace protectForm mixin by a higher order component

While working on GeneralForm in SiteSettings, I could not refactor the component to ES6 because of the`protectForm`. So I thought it's a good opportunity to get rid of this mixin and replace it with a HoC.

I kept it working exactly the same way as the mixin, a global variable of active dirty forms that we check on route exit. The only difference is that we no longer call `this.markSaved` and `this.markChanged` but we use props instead `this.props.markSaved` and `this.props.markChanged`.

I had to change all the components using this mixin to avoid having two separate variables storing dirtyForms.

**Testing instructions**

It's used in a couple of places, so make sure these forms trigger the confirmation popup if we try to leave without saving:

 * All settings forms from the "Settings" menu
 * The user profile form
 * The account settings form
 * The change password form
 * The 2FA form
 * The post editor
 * The notifications form for reader subscriptions
 * The edit team member form (from the people menu)
 * The sharing buttons form
 
cc @aduth @timmyc
